roofit/roofitcore/src/RooLinearVar.cxx

```diff
@@ -248,9 +248,7 @@ std::list<std::string> RooLinearVar::getBinningNames() const
 {
   std::list<std::string> binningNames(1, "");
 
-  RooFIter iter = _altBinning.fwdIterator();
-  const RooAbsArg* binning = 0;
-  while((binning = iter.next())) {
+  for (RooAbsArg const* binning : _altBinning) {
```

Comment:
```suggestion
  for (TObject const* binning : _altBinning) {
```
The code doesn't compile now, as the type is not `RooAbsArg` but `TObject`. I think just changing this should fix it (sorry for suggesting the wrong type in the previous comment!).